e2e/scripts/smoke_test.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash -e
+
+test_source_repo=$(pwd)/e2e
+docker_test_repo=/workspace/go/src/github.com/rook/rook/e2e
+git_smoke_test_directory=github.com/rook/rook/e2e/tests/smoke
+container_image=quay.io/quantum/rook-test
+tmp_docker_sock_path=/tmp/docker.sock
+results_dir=results
+set_go_env_var="export GOPATH=/workspace/go && export GOROOT=/usr/lib/go"
+
+#create the rook infrastructure container
+rook_infra::create() {
+    export id=$(docker run \
+        -itd \
+        --network=host \
+        -e container=docker \
+        --privileged \
+        --security-opt=seccomp:unconfined \
+        --cap-add=SYS_ADMIN \
+        -v /dev:/dev \
+        -v /sys:/sys \
+        -v /sys/fs/cgroup:/sys/fs/cgroup \
+        -v /sbin/modprobe:/sbin/modprobe \
+        -v /lib/modules:/lib/modules:rw \
+        -v /var/run/docker.sock:${tmp_docker_sock_path} \
+        -v ${test_source_repo}:/${docker_test_repo} \
+        -p 8080 \
+        -w ${docker_test_repo} \
+        ${container_image} \
+        /sbin/init)
+
+        rc=$?; if [[ $rc != 0 ]]; then exit $rc; fi
+}
+
+#prepare the rook_infra container to run tests
+rook_infra::init() {
+    echo Mounting host docker.sock...
+
+    docker exec -i ${id} rm -rfv /var/run/docker.sock
+    rc=$?; if [[ $rc != 0 ]]; then exit $rc; fi
+
+    docker exec -i ${id} ln -s ${tmp_docker_sock_path} /var/run/docker.sock
+    rc=$?; if [[ $rc != 0 ]]; then exit $rc; fi
+    echo Success...
+
+    echo Installing rook-test-framework dependencies...
+    docker exec -i ${id} /bin/bash -c \
+    "${set_go_env_var} && glide install"
+    rc=$?; if [[ $rc != 0 ]]; then exit $rc; fi
+    echo Success...
+}
+
+rook_infra::run_test() {
+    test_name_regex=$1
+
+    #TODO:: pipe it to junit parser
+    #TODO:: get exit code and fail if not 0
+
+    docker exec ${id} /bin/bash -c \
+   "${set_go_env_var} && go test -run ${test_name_regex} ${git_smoke_test_directory} -v"
```

Comment:
it might be nice if the caller of this script can pass args to it that would then be passed through here to the `go test` run.  Flags such as `--k8s_version` and `--rook_tag`.

You could consider just making this call with a `$@` meaning the rest of the args passed into `smoke_test.sh`